blockly/src/utils/BlocklyCommands.java

```diff
@@ -186,29 +188,35 @@ public static void pull() {
    * Shoots a fireball in direction the hero is facing.
    *
    * @param ac AmmunitionComponent of the hero, ammunition amount will be reduced by 1
+   * @param hero Entity to be used as hero for positioning
    */
-  private static void aimAndShoot(AmmunitionComponent ac) {
-    Entity hero = Game.hero().orElseThrow(MissingHeroException::new);
-    Direction viewDirection =
-        Direction.fromPositionCompDirection(EntityUtils.getViewDirection(hero));
-    Skill fireball =
-        new Skill(
-            new FireballSkill(
-                () ->
-                    hero.fetch(CollideComponent.class)
-                        .map(cc -> cc.center(hero))
-                        .map(p -> p.add(viewDirection.toPoint()))
-                        .orElseThrow(
-                            () -> MissingComponentException.build(hero, CollideComponent.class)),
-                FIREBALL_RANGE,
-                FIREBALL_SPEED,
-                FIREBALL_DMG),
-            1);
-    fireball.execute(hero);
+  private static void aimAndShoot(AmmunitionComponent ac, Entity hero) {
+    newFireballSkill(hero).execute(hero);
     ac.spendAmmo();
     Server.waitDelta();
   }
 
+  /**
+   * Create a new fireball for the given entity.
+   *
+   * @param hero Entity to be used as hero for positioning
+   * @return Nice new fireball, ready to be launched.
+   */
+  private static Skill newFireballSkill(Entity hero) {
```

Comment:
```suggestion
  private static Skill newFireballSkill() {
```
eigentlich kannste den Hero hier als Parameter doch weglassen, die BlocklyCommands werden immer vom helden ausgeführt, ist also implizit.
Dann würdest du dir auch oben dein 2x`hero` ersparen.

Reply:
Wäre mir tatsächlich lieb, aber zwei Zeilen tiefer will ich die CollideComponent vom Hero haben und auch die Position und so und brauche ihn für die Exception:

```java
  private static Skill newFireballSkill(Entity hero) {
    return new Skill(
        new FireballSkill(
            () ->
                hero.fetch(CollideComponent.class)
                    .map(cc -> cc.center(hero))
                    .map(p -> p.add(Direction.asPoint(EntityUtils.getViewDirection(hero))))
                    .orElseThrow(
                        () -> MissingComponentException.build(hero, CollideComponent.class)),
                    ...
```

Und den Hero hatte ich ja bereits aus Game geholt, nur nicht hier. D.h. entweder hole ich ihn hier nochmal neu (was ich eigentlich vermeiden wollte) oder ich muss ihn übergeben.

Reply:
@AMatutat @Flamtky Ich habe die Idee in #2008 mal ausprobiert. Könnte das so gehen?

Reply:
OK. Das war wirklich zu simpel ... D.h. das ständige Fetchen des Helden ist noch mit die lesbarere und verständlichere Option. Dann kann hier zu.